pyproject.toml

```
@@ -64,6 +56,31 @@ warn_unused_ignores = true
 warn_redundant_casts = true
 warn_unused_configs = true
 
+[tool.ruff]
+# Exclude a variety of commonly ignored directories.
+exclude = ["docs*", "conda*"]
+
+[tool.ruff.lint]
+# E4 - whitespace
+# E7 - multiple-statements
+# E9 - trailing-whitespace
+# F - Enable Pyflakes
+# B - Enable flake8-bugbear
+# W - Enable pycodestyle
+# C901 - complex-structure
+# D - Enable flake8-docstrings
+select = ["E4", "E7", "E9", "F", "B", "W", "C901"]
+
+# E501 - line-too-long
+ignore = ["E501"]
+
```

Let's add:
```suggestion

[tool.ruff.format]
quote-style = "single"
```
This will enforce single quotes, rather than double quotes.